components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.server/src/main/java/org/wso2/carbon/identity/configuration/mgt/server/cors/CORSManagementService.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.configuration.mgt.server.cors;
+
+import org.wso2.carbon.identity.configuration.mgt.server.cors.exception.CORSServiceException;
+
+import java.util.List;
+
+/**
+ * Service for managing the CORS URLs of a tenant.
+ */
+public interface CORSManagementService {
+
+    /**
+     * Get all the CORS URLs belonging to a tenant.
+     *
+     * @param tenantDomain The tenant domain.
+     * @return List<String> Returns a list of CORS URLs configured by the tenant as strings.
+     * @throws CORSServiceException
+     */
+    List<String> getCORSUrls(String tenantDomain) throws CORSServiceException;
+
+    /**
+     * Set the CORS URLs for a tenant. This method replaces any existing URLs.
+     *
+     * @param tenantDomain The tenant domain.
+     * @param urls A list of CORS URLs.
+     * @throws CORSServiceException
+     */
+    void setCORSUrls(String tenantDomain, List<String> urls) throws CORSServiceException;
+
+    /**
+     * Add the CORS URL(s) to the existing URL list of the tenant.
+     *
+     * @param tenantDomain The tenant domain.
+     * @param urls A list of CORS URLs.
+     * @throws CORSServiceException
+     */
+    void addCORSUrls(String tenantDomain, List<String> urls) throws CORSServiceException;
```

Comment:
Add usually denotes and append.
It's fine to pass the list, but we should have another method to add just one origin. WDYT?

Reply:
I had that and removed it since extra method seemed redundant when the method to add multiple origins can be used to add one origin as well.

Reply:
This method appends the list of new origins rather than replacing existing ones.